internal/xds/translator/cluster.go

```diff
@@ -61,10 +60,24 @@ func buildXdsCluster(clusterName string, tSocket *corev3.TransportSocket, protoc
 		cluster.RespectDnsTtl = true
 	}
 
-	if protocol == HTTP2 {
+	if args.protocol == HTTP2 {
 		cluster.TypedExtensionProtocolOptions = buildTypedExtensionProtocolOptions()
 	}
 
+	// Set Load Balancer policy
+	//nolint:gocritic
+	if args.loadBalancer == nil {
+		cluster.LbPolicy = clusterv3.Cluster_LEAST_REQUEST
+	} else if args.loadBalancer.LeastRequest != nil {
+		cluster.LbPolicy = clusterv3.Cluster_LEAST_REQUEST
+	} else if args.loadBalancer.RoundRobin != nil {
+		cluster.LbPolicy = clusterv3.Cluster_ROUND_ROBIN
+	} else if args.loadBalancer.Random != nil {
+		cluster.LbPolicy = clusterv3.Cluster_RANDOM
+	} else if args.loadBalancer.ConsistentHash != nil {
+		cluster.LbPolicy = clusterv3.Cluster_MAGLEV
```

Comment:
Emissary supports maglev and ring hash to allow for session affinity based on request headers, auto-generated cookies, and source IP if we want to design around feature parity. The IR can always change so this just just a consideration for the public API for this.

Reply:
imo is there any use case to support ring hash over maglev ? afaik maglev is superior to ring hash when it comes down to consistent hashing

Reply:
ive only added only hash input - source ip, we should be able to add other such as cookie etc in the future

Reply:
I think that's totally fine unless someone has a reason to request support for ring hash, then we can consider it at that point.